### ISSUE TITLE
Outdated build instruction on OS X Yosemite

### ISSUE THREAD

**Issue by lacombar** [symless/synergy#4572]
The build instruction provided on the [website](http://synergy-project.org/wiki/Compiling#Mac_OS_X_10.10_Yosemite) no longer works:

```
$ patch -p1 < mac-synergy.patch
patching file ext/toolchain/commands1.py
Hunk #1 succeeded at 432 (offset -19 lines).
Hunk #2 FAILED at 467.
Hunk #3 succeeded at 551 (offset -27 lines).
1 out of 3 hunks FAILED -- saving rejects to file ext/toolchain/commands1.py.rej
```

I suppose this is due to the recent change in `ext/toolchain/commands1.py` made in  3deb5492d60f2f181ea42aa0ff5b98410b29a7c4

The following (simple) change seems to be working fine for me:

```
diff --git a/ext/toolchain/commands1.py b/ext/toolchain/commands1.py
index 484b9da..d141ca8 100644
--- a/ext/toolchain/commands1.py
+++ b/ext/toolchain/commands1.py
@@ -431,7 +431,7 @@ from getopt import gnu_getopt
                if generator.cmakeName.find('Unix Makefiles') != -1:
                        cmake_args += ' -DCMAKE_BUILD_TYPE=' + target.capitalize()

-               elif sys.platform == "darwin":
+               if sys.platform == "darwin":
                        macSdkMatch = re.match("(\d+)\.(\d+)", self.macSdk)
                        if not macSdkMatch:
                                raise Exception("unknown osx version: " + self.macSdk)
```

**Comment by VerbGarden** [comment]
I have also run into this precise error in patching as of today around 11AM MST.

**Comment by floorjoist** [comment]
I also have this problem - couldn't get the listed patch to work. (copy and pasted and tried to apply this patch after the first patch. i didn't think you could patch the patch..)

**Comment by kanjames** [comment]
Sorry I'm a newb, but where do you put this code? And how do you execute it?

I ran this first, but ran into the same problem:
patch -p1 < mac-synergy.patch 

patching file ext/toolchain/commands1.py
Hunk #1 succeeded at 431 (offset -20 lines).
Hunk #2 FAILED at 466.
Hunk #3 succeeded at 550 (offset -28 lines).
1 out of 3 hunks FAILED -- saving rejects to file ext/toolchain/commands1.py.rej

MAIN QUESTION:
I put it as a new file and ran it with the command:
 patch -p1 < mac-synergy2.patch 

But received the following result:
patching file ext/toolchain/commands1.py
Hunk #1 FAILED at 431.
1 out of 1 hunk FAILED -- saving rejects to file ext/toolchain/commands1.py.rej

Any help would be greatly appreciated

**Comment by corinnaSchultz** [comment]
kanjames, if you open the file in a text editor, ext/toolchain/commands1.py and find line 433 (which says: elif sys.platform == "darwin": ) and change the "elif" to an "if" and then save the file, you'll be fine. If you used brew to install qt, you will still need to make the other change mentioned in the wiki.

line 774 in the same file, ext/toolchain/commands1.py , change the path to this (assuming that is the correct path for you):
frameworkRootDir = "/usr/local/Cellar/qt/4.8.7/Frameworks"

**Comment by corinnaSchultz** [comment]
A general question I have, is whether it's ok to just change the elif to an if? Is there any problem with having those cmake options there, which were excluded on the original patch: 

-DCMAKE_OSX_SYSROOT
-DCMAKE_OSX_DEPLOYMENT_TARGET

**Comment by lacombar** [comment]
Are we really discussing the color of the bikeshed ?

If this works, then, it's OK.

**Comment by corinnaSchultz** [comment]
Since I don't know what I'm doing, I'm hesitant to say this works, or to advise other people to do this. I'm a total newbie when it comes to cmake and building on mac via command line. I'd rather not dig into cmake documentation is I don't have to.

**Comment by lacombar** [comment]
Few questions:

Does synergy build on Yosemite as-is, from a fresh clone ?

Does synergy build on Yosemite with the "semi-official" patch ?

Does synergy build on Yosemite with this patch ?

**Comment by corinnaSchultz** [comment]
Fresh clone: configure fails, so I assume a build will fail too

Semi-official patch: configure succeeds, once you figure out how to apply the patch by hand. I didn't try building

Simply changing elif to if: configure succeeds, there are a number of warnings during build, and then there's a failure related to a bad Qt Library path, even though I fixed that one path mentioned on the wiki. I'll post here after I track down the cause and get a successful build.

**Comment by corinnaSchultz** [comment]
OK, it builds successfully. I put the framework path in the wrong place, oops.

kanjames, replace both paths with your actual Qt framework path, so that frameworkRootDir will be sure to point to the right directory.

**Comment by floorjoist** [comment]
I've confirmed these changes work for me. I do notice that the qt window
for 1.7.4 doesn't look as polished as it did in 1.4.x - the lines of text
aren't centered in their areas, and it's not as nicely formatted.

oh well, it works now!

**Comment by moebiussurfing** [comment]
Afterdoing this changes, I am getting this error when building:

 ./hm.sh conf -g1 --mac-sdk 10.10 --mac-identity test
Mapping command: conf -> configure
Running setup...
Setup complete.
Error: Mac SDK not found at: 2015-07-29 16:45:44.953 xcodebuild[28516:1327680] [MT] PluginLoading: Required plug-in compatibility UUID 7FDF5C7A-131F-4ABB-9EDC-8C5F8F0B8A90 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/CocoaPodUI.xcplugin' not present in DVTPlugInCompatibilityUUIDs
2015-07-29 16:45:44.954 xcodebuild[28516:1327680] Failed to load plugin at: /Users/manu/Library/Application Support/Developer/Shared/Xcode/Plug-ins/CocoaPodUI.xcplugin, skipping.  Reason for failure: *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.10.sdk

**Comment by corinnaSchultz** [comment]
Perhaps your version of XCode isn't up-to-date? See this thread, where someone complains about a similar issue: http://stackoverflow.com/questions/11989007/where-to-get-macos-sdk-10-6-for-xcode

**Comment by moebiussurfing** [comment]
@corinnaSchultz yes, it's up to date... 10.10.4 & Xcode 6.4
anywa I donloaded the free version from here https://synergy-project.org/nightly
PS: do you know if there's some limitation without the "not free"/"own-compiled" and registered version?

**Comment by moebiussurfing** [comment]
... answering our reply: 10.10 sdk is there. but maybe it needs 10.6 like on the link?
i think this is related to the error:  http://stackoverflow.com/questions/20732327/xcode-5-required-plug-in-not-present-in-dvtplugincompatibilityuuids

**Comment by corinnaSchultz** [comment]
Yeah try adding the UUID from the error message into the plist file. If that works, leave a comment here.

**Comment by moebiussurfing** [comment]
... I can't see where is the Xcode project.. sorry

**Comment by corinnaSchultz** [comment]
Did you look in synergy/build? I think the file was created for you by hm.sh

**Comment by moebiussurfing** [comment]
hmm no, it stopped working before create it… Anyway I can use the compiled free version by now.

**Comment by hstdt** [comment]
@corinnaSchultz 

Here's my rough way to make it compilable.
(ENV : OS X 10.10.5, Xcode 7.2.1)

1.command1.py
![image](https://cloud.githubusercontent.com/assets/10215098/13976406/70881fe2-f0fa-11e5-8003-4bb917475934.png)

`self.getMacSdkDir()` should return `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk` but '/Developer/SDKs/MacOSX10.10.sdk`

with an error:

![image](https://cloud.githubusercontent.com/assets/10215098/13973855/e75e591a-f0e1-11e5-934d-ebc64beacbb4.png)

It is said that sdk path has been changed. [stackoverflow 2009555](http://stackoverflow.com/questions/2009555/iphone-sdk-location-on-hard-drive)

The folder /Developer is not exist.

I'm not sure whether `xcrun` is a good way to get SDK path, Xcode-plugins in `/Users/Your User Name/Library/Application Support/Developer/Shared/Xcode/Plug-ins` may cause error:
![image](https://cloud.githubusercontent.com/assets/10215098/13974002/2117d7ac-f0e3-11e5-8223-1689dd60dc68.png)
We need move them to another folder before running.

But It still returns '/Developer/SDKs/MacOSX10.10.sdk` after remove all plugins, so I have to directly change sdkDir to `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk`

`getMacSdkDir(self)` may have some logical problems, but I have never learnt about python...

2.CMakeLists.txt

to avoid the error:

![image](https://cloud.githubusercontent.com/assets/10215098/13976732/bbbb1fe8-f0fd-11e5-957d-25578cab9af9.png)

directly run:

![image](https://cloud.githubusercontent.com/assets/10215098/13976628/91bb3c92-f0fc-11e5-8191-d238b013b8ba.png)

That's all.

Now,It's time to run the code below.
```
 ./hm.sh conf -g1 [-d] # Use -d to build a debug version.
 ./hm.sh build [-d]
```

**Comment by Macintoshleong** [comment]
I am experiencing a weird problem when compiling after manage to link cmake ,
![screen shot 2016-04-23 at 4 37 37 pm](https://cloud.githubusercontent.com/assets/6150058/14773291/f0857a56-0ada-11e6-93d8-88b3c00b9e9b.png)
Pls help..

**Comment by Macintoshleong** [comment]
I do not know what to do...any effective way to solve this?

**Comment by martinweiss** [comment]
I have the same problem as Macintoshleong. Any help would be greatly appreciated.

**Comment by Macintoshleong** [comment]
after a month someone reply i feel u bro

**Comment by xapple** [comment]
I used the instructions found here: 

https://github.com/symless/synergy/wiki/Compiling#Mac_OS_X_1010_and_above

and it worked perfectly on 10.11.5. Just letting you know.

**Comment by Macintoshleong** [comment]
Thx @xapple will try this method tomorrow.

**Comment by Macintoshleong** [comment]
@xapple r u using the Mac OS X 10.10 and above method are below ?, can u state your state?

**Comment by narainabhi** [comment]
I am using qt 5.8 on Mac Sierra. I did follow the instructions at 
https://github.com/symless/synergy/wiki/Compiling#Mac_OS_X_1010_and_above

Getting an error : Destination path '5/Resources' already exists.
Anyone else getting this ?

**Comment by karlTbowers** [comment]
Hi narainabhi,
I am getting the same error. qt 5.8 on Sierra. I have not been able to fix it yet.
The Synergy app will launch, but no gui elements on the actual windows.
I tried doing a clean then build again. That did not help.

**Comment by narainabhi** [comment]
any one else has a fix for the above issue ?

**Comment by egzumer** [comment]
Same problem here. Hoping for fix :)

**Comment by giovannicandido** [comment]
Me to Error: Destination path '5/Resources' already exists

**Comment by egzumer** [comment]
The error is from fixQtFrameworksLayout, which does basically nothing. It moves some files that are already where they need to be and that causes the "already exists" error. This function call needs to removed. But there is another problem. Application starts but all forms (windows) are empty.

**Comment by jersonperpetua** [comment]
Follow @egzumer's steps, and try **qt@5.5** from Homebrew:

```
brew install qt@5.5
```

Hope it works for you too!

**Comment by egzumer** [comment]
Finally I managed to build it. I had Qt from homebrew (5.8) and Qt 5.2.1 from qt archive. No need to erase fixQtFrameworksLayout. I did:
export PATH=$PATH:/usr/local/Cellar/qt5/5.8.0_1/bin
python hm.py conf -g2 --mac-sdk 10.12 --mac-identity test
--reopen new console--
export PATH=$PATH:/Users/user/Qt5.2.1/5.2.1/clang_64/bin
python hm.py build
python hm.py package mac

**Comment by ravenium** [comment]
This still feels a little clunky, having to grab archival Qt's.  I figured out a workaround, but downside is that it's not a static compile:  https://github.com/symless/synergy/issues/5994

**Comment by sk9c00** [comment]
So, looks like the issue still hasn't been fixed, but this is just a retest from scratch on an up-to-date system.

macOS Sierra 10.12.5 w/ Xcode 8.1

Using HomeBrew to attain:
`brew info qt
qt: stable 5.9.0 (bottled), HEAD [keg-only]
Cross-platform application and UI framework
https://www.qt.io/
/usr/local/Cellar/qt/5.9.0 (9,015 files, 296.3MB)
  Poured from bottle on 2017-06-02 at 11:00:05
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/qt.rb`
`brew info cmake
cmake: stable 3.8.2 (bottled), HEAD
Cross-platform make
https://www.cmake.org/
/usr/local/Cellar/cmake/3.8.2 (2,208 files, 29.3MB) *
  Poured from bottle on 2017-06-02 at 10:57:22
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/cmake.rb`

TL;DR Steps:
`brew install qt
brew install cmake

#To eliminate git related errors.
git clone https://github.com/symless/synergy.git

#To get qt version
brew info qt 
#To fix any qmake path dependancy issue
export PATH=/usr/local/Cellar/qt/5.9.0/bin:$PATH

Changing in commands1.py
/usr/local/Cellar/qt/5.9.0/Frameworks

#Changing /Content/ plist to /Resources/ plist in commands.py creates
#Error: Destination path '5/Resources' already exists

#Attempted Build 1
./hm.sh conf -g2 --mac-sdk 10.12 --mac-identity test
./hm.sh build

# Build Attempt 2, after -g2, this magically works alone.
./hm.sh conf -g1
./hm.sh build`

Pretty much creates a Synergy.app which is executable, but like all the people above, empty GUI elements. On the plus side,
`./hm.sh dist mac`
works.

**Comment by philion** [comment]
I can't get around:

Going back to: <path-redacted>/synergy
bin/Synergy.app
Entering dir: bin/Synergy.app/Contents/Frameworks/QtCore.framework/Versions
Error: Destination path '5/Resources' already exists

A note to maintainers: I was planning on buying a license, but how can I trust software I can't even build?

**Comment by yupi2** [comment]
@philion If you can't trust software you can't build then why do you use macOS? Anyway try #5826 out.

**Comment by revolter** [comment]
You can't trust an open source software that you can't build, not closed source.

**Comment by tb0hdan** [comment]
Here's working (so far) Makefile for OSX Sierra: https://github.com/tb0hdan/synergy_build_osx

**Comment by revolter** [comment]
@tb0hdan, What about #5905. Isn't that happening to you too?

**Comment by bofish** [comment]
@sk9c00 , I do not understand the step of "Changing /Content/ plist to /Resources/ plist in commands.py creates".

Should I change the content in commands.py? But I don't find any file named commands.py in synergy folder, there is only commands1.py.

**Comment by nbolton** [comment]
This issue has been inactive for a while so I'll close it for now. 
Please let me know if this is still an issue so that I can reopen it.
